### PR TITLE
New rake task to delete re-saved users

### PR DESCRIPTION
**Pull request by MarcReniu**
#### :tophat: What? Why?
Re-saved users after a DB backup restore must be deleted again

#### :pushpin: Related Issues
- Related to Epic #3320
- Relates to issue #3315

#### :clipboard: Subtasks
- [] Add `CHANGELOG` entry
- [ ] Add documentation regarding the feature 
- [ ] Add/modify seeds
- [X ] Add tests
- [ ] Another subtask

### CODE REVIEW

File: decidim-core/lib/tasks/decidim_tasks.rake [hunk 1]
```diff
@@ -3,4 +3,36 @@
 namespace :decidim do
   desc "Install migrations from Decidim to the app."
   task upgrade: ["railties:install:migrations"]
+
+  task right_to_be_forgotten: :environment do
```
Comment (agustibr) [inline decidim-core/lib/tasks/decidim_tasks.rake]: I would add a meaningful and simple description that describes the task

File: decidim-core/lib/tasks/decidim_tasks.rake [hunk 2]
```diff
@@ -3,4 +3,36 @@
...
+    begin
+      puts "RightToBeForgotten: -------------------"
+      path = ENV["FILE_PATH"].presence || "tmp/forgotten_users.csv" # Good file
+      file_path = Rails.root.join(path)
+      CSV.foreach(file_path, headers: false) do |row|
+        # Each row, first element must be ID, others are useless
```
Comment (agustibr) [inline decidim-core/lib/tasks/decidim_tasks.rake]: Should this task have a log? 
```
log = ActiveSupport::Logger.new('log/right_to_be_forgotten.log')
```

File: decidim-core/lib/tasks/decidim_tasks.rake [hunk 3]
```diff
@@ -3,4 +3,36 @@
 namespace :decidim do
   desc "Install migrations from Decidim to the app."
   task upgrade: ["railties:install:migrations"]
+
+  task right_to_be_forgotten: :environment do
+    begin
+      puts "RightToBeForgotten: -------------------"
```
Comment (agustibr) [inline decidim-core/lib/tasks/decidim_tasks.rake]: I would add a `#{start_time}`

File: decidim-core/lib/tasks/decidim_tasks.rake [hunk 4]
```diff
@@ -3,4 +3,36 @@
...
+        end
+      end
+    rescue CSV::MalformedCSVError
+      puts " ERROR: [Malformed CSV] #{path}"
+    rescue Errno::ENOENT
+      puts " ERROR: [File not found] #{path}"
+    rescue StandardError => e
+      puts " ERROR: [Unexpected error] #{e.message}"
+    ensure
+      puts "RightToBeForgotten: --------------- END"
```
Comment (agustibr) [inline decidim-core/lib/tasks/decidim_tasks.rake]: I would add a `#{end_time}`

File: decidim-core/lib/tasks/decidim_tasks.rake [hunk 5]
```diff
@@ -3,4 +3,36 @@
 namespace :decidim do
   desc "Install migrations from Decidim to the app."
   task upgrade: ["railties:install:migrations"]
+
+  task right_to_be_forgotten: :environment do
+    begin
+      puts "RightToBeForgotten: -------------------"
+      path = ENV["FILE_PATH"].presence || "tmp/forgotten_users.csv" # Good file
+      file_path = Rails.root.join(path)
+      CSV.foreach(file_path, headers: false) do |row|
```
Comment (agustibr) [inline decidim-core/lib/tasks/decidim_tasks.rake]: I would add a `users_count`

File: decidim-core/lib/tasks/decidim_tasks.rake [hunk 6]
```diff
@@ -3,4 +3,36 @@
...
+        # Each row, first element must be ID, others are useless
+        user_id = row.first.presence || ""
+        next unless user_id =~ /^\d+$/ # ID must be numeric only
+
+        if (user = Decidim::User.find_by(id: user_id))
+          if user.deleted?
+            puts "  - [#{user_id}] User already deleted"
```
Comment (agustibr) [inline decidim-core/lib/tasks/decidim_tasks.rake]: I'll add to the log `log.info "#{index}/#{users_count} - [#{user_id}] User already deleted"`

File: decidim-core/lib/tasks/decidim_tasks.rake (hunk 5 above)
Comment (MarcReniu) [reply decidim-core/lib/tasks/decidim_tasks.rake]: Why? I mean, each user_id managed have his own output.

File: decidim-core/lib/tasks/decidim_tasks.rake [hunk 7]
```diff
@@ -3,4 +3,36 @@
...
+          else
+            puts " !! [#{user_id}] DELETING USER"
```
Comment (agustibr) [inline decidim-core/lib/tasks/decidim_tasks.rake]: I'll add to the log `log.info "#{index}/#{users_count} - [#{user_id}] DELETING USER"`

File: decidim-core/lib/tasks/decidim_tasks.rake [hunk 8]
```diff
@@ -3,4 +3,36 @@
...
+            Decidim::DestroyAccount.call(user, Decidim::DeleteAccountForm.from_params({})) # Delete user!
+          end
+        else
+          puts "  - [#{user_id}] User not found"
```
Comment (agustibr) [inline decidim-core/lib/tasks/decidim_tasks.rake]: I'll add to the log log.info "#{index}/#{users_count} - [#{user_id}] User not found"

**Review by agustibr — APPROVED** [review]
:+1:  great!

**Pull request closed without merging**